### ISSUE TITLE
Vanished players are able to be messaged with ChatControlRed

### ISSUE THREAD

**Issue by RhinoCD** [quantiom/AdvancedVanish#15]
![image](https://user-images.githubusercontent.com/78483451/183969318-24d0ec87-0f30-4ff9-b52f-d73be62f5f1f.png)

Maybe a way to prevent players from messaging someone in vanish? / running commands on players who are in vanish

**Comment by RhinoCD** [comment]
Same with things like /trade, etc.

**Comment by quantiom** [comment]
It's only possible to do this if those specific plugins add checks for vanished players or it can be added as a hook to AdvancedVanish. Which plugins do you use for messaging and trading?

**Comment by RhinoCD** [comment]
ChatControlRed - I don’t know if trading actually works or has checks ill have to verify, but messaging I guess is the important one. Trading would be TradeSystems

**Comment by quantiom** [comment]
I don't think it will be possible for me to add a hook for that plugin. On the [API page of the plugin's wiki](https://github.com/kangarko/ChatControl-Pro/wiki/Developer-API), it says to PM "kangarko" on Spigot for a free developer license, but it looks like he's been banned.

Also, to use the API, it says: 
> using the systemPath dependency scope in Maven or importing as a library in your build path,

which would not be possible for me to add to AdvancedVanish without distributing the paid plugin for free.

I would recommend opening an issue there and probably referencing this issue in it.

**Comment by RhinoCD** [comment]
Okay! tysm

**Comment by RhinoCD** [comment]
Heres the link to the page if needed too: https://builtbybit.com/resources/chatcontrol-red-format-filter-your-chat-1-chat-plugin-1-2-5-1-19-1-10-off.18217/

**Comment by ElBananaa** [comment]
Hey,
I did try to work on implementing support for your plugin, however it seems like I am completely unable to get the plugin using jitpack.
Is there an alternative repository I can get the plugin from?

**Comment by quantiom** [comment]
> Hey, I did try to work on implementing support for your plugin, however it seems like I am completely unable to get the plugin using jitpack. (see https://jitpack.io/#quantiom/AdvancedVanish) Is there an alternative repository I can get the plugin from?

It seems like the project won't build on jitpack anymore, I think because of the network speed. I switched the repository to repsy and updated the instructions in the README. Let me know if it works for you.

**Comment by ElBananaa** [comment]
> > Hey, I did try to work on implementing support for your plugin, however it seems like I am completely unable to get the plugin using jitpack. (see https://jitpack.io/#quantiom/AdvancedVanish) Is there an alternative repository I can get the plugin from?
> 
> It seems like the project won't build on jitpack anymore, I think because of the network speed. I switched the repository to repsy and updated the instructions in the README. Let me know if it works for you.

 Awesome, thanks! It works fine now :)

**Comment by quantiom** [comment]
> > > Hey, I did try to work on implementing support for your plugin, however it seems like I am completely unable to get the plugin using jitpack. (see https://jitpack.io/#quantiom/AdvancedVanish) Is there an alternative repository I can get the plugin from?
> > 
> > 
> > It seems like the project won't build on jitpack anymore, I think because of the network speed. I switched the repository to repsy and updated the instructions in the README. Let me know if it works for you.
> 
> Awesome, thanks! It works fine now :)
> 
> Btw, if you wish to get access to chatcontrol in order to hook with its api, feel free to send an email to kangarko [matej@mineacademy.org](mailto:matej@mineacademy.org). I believe he also plans to release the api publicly but I don't think he gave an eta for this.

If I were to do that, I would not be allowed to distribute it and other developers would not be able to build AdvancedVanish.

Also, I see that support has been added (https://github.com/kangarko/ChatControl-Red/issues/2085), so I will be closing this as completed. Thanks!

**Comment by ElBananaa** [comment]
> > > > Hey, I did try to work on implementing support for your plugin, however it seems like I am completely unable to get the plugin using jitpack. (see https://jitpack.io/#quantiom/AdvancedVanish) Is there an alternative repository I can get the plugin from?
> > > 
> > > 
> > > It seems like the project won't build on jitpack anymore, I think because of the network speed. I switched the repository to repsy and updated the instructions in the README. Let me know if it works for you.
> > 
> > 
> > Awesome, thanks! It works fine now :)
> > Btw, if you wish to get access to chatcontrol in order to hook with its api, feel free to send an email to kangarko [matej@mineacademy.org](mailto:matej@mineacademy.org). I believe he also plans to release the api publicly but I don't think he gave an eta for this.
> 
> If I were to do that, I would not be allowed to distribute it and other developers would not be able to build AdvancedVanish.
> 
> Also, I see that support has been added ([kangarko/ChatControl-Red#2085](https://github.com/kangarko/ChatControl-Red/issues/2085)), so I will be closing this as completed. Thanks!

Alright, thanks to you!
I guess publishing the api is something Kangarko needs to work on quickly ^^
